xbrowse_server/phenotips/views.py

```diff
@@ -1,90 +1,187 @@
 from django.shortcuts import render
 from django.http import JsonResponse
 from django.contrib.auth.decorators import login_required
-import urllib2
-import base64
 from django.conf import settings
 import datetime
 import time
 import os
 from xbrowse_server.decorators import log_request
 import ast
-import json
 import logging
+from django.http.response import HttpResponse
+from django.http import Http404
+from django.views.decorators.csrf import csrf_exempt
+import requests
+from xbrowse_server.phenotips.utilities import do_authenticated_call_to_phenotips
+from xbrowse_server.phenotips.utilities import convert_internal_id_to_external_id
+from xbrowse_server.phenotips.utilities import get_uname_pwd_for_project
+from xbrowse_server.phenotips.utilities import get_generic_collaborator_uname_pwd_for_project
+
+from xbrowse_server.base.models import Project
+from django.shortcuts import render, redirect, get_object_or_404
+  
 logger = logging.getLogger(__name__)
 
-@log_request('sync_data_api')
+  
+@log_request('phenotips_proxy_edit_page')
 @login_required
-#handle a request to sync data between xbrowse and phenotips
-def process_sync_request(request):
-  '''handle a request to sync data'''
-  message={'message':'meant for AJAX POST'}
-  if request.method == 'POST':
-      if request.is_ajax():
-          uname = request.POST.get('uname')
-          pwd = request.POST.get('pwd')
-          individual_id=request.POST.get('id')
-          if __process_sync_request_helper(uname,pwd,individual_id):
-            message={'status':'success'}
-          else:
-            message={'status':'error'}
-  return JsonResponse(message)
+#test function to see if we can proxy phenotips
+#exempting csrf here since phenotips doesn't have this support
+@csrf_exempt
+def fetch_phenotips_edit_page(request,eid):
+  '''test function to see if we can proxy phenotips'''  
+  current_user = request.user
+  
+  if request.GET.has_key('project'):
+    project_name=request.GET['project']  
+    #add project name to session for use in proxying
+    request.session['current_project_name']=project_name
```

Comment:
thanks! Sessions came in pretty handy. In fact we might have to rely on this at some point to speed  up our system by minimizing Phenotip login connections to 1 per session.